emanate/__init__.py

```diff
@@ -134,12 +78,50 @@ def del_symlink(self, path_obj):
         return not dest_file.exists()
 
     def run(self):
-        all_files = Path(".").glob("**/*")
+        all_files = Path(self.config.source).glob("**/*")
         files = filter(self.valid_file, all_files)
         list(map(self.function, files))
 
+
+def parse_args(args=None):
+    argparser = ArgumentParser(
+        description="symlink files from one directory to another",
+        argument_default=SUPPRESS
+    )
+    argparser.add_argument("--clean", help="Remove symlinks.")
+    argparser.add_argument("--destination",
+                           metavar="DESTINATION",
+                           help="Directory to create and remove symlinks in.")
+    argparser.add_argument("--source",
+                           metavar="SOURCE",
+                           type=Path,
+                           help="Directory holding the files to symlink.")
+    argparser.add_argument("--no-confirm",
+                           action="store_false",
+                           dest="confirm",
+                           help="Don't prompt before replacing a file.")
+    argparser.add_argument("--config",
+                           metavar="CONFIG_FILE",
+                           default=None,
+                           type=Path,
+                           help="Configuration file to use.")
+
+    return argparser.parse_args(args)
+
+
 def main():
-    return Emanate(sys.argv).run()
+    """Emanate prioritizes configuration sources in the following order:
+    - default values have lowest priority;
+    - the configuration file overrides defaults;
+    - command-line arguments override everything."""
+    args = parse_args()
+    if args.config is None:
+        args.config = args.source / "emanate.json"
+
+    return Emanate(
+        config.from_json(args.config) if args.config.exists() else None,
```

Comment:
What're your thoughts on providing something like `config.try_from_json(...)` which includes returns `None` if `args.config.exists()` is `False`?

I'm unsure how I feel about it, but this seems something that would be a common code snippet when it's used.

Reply:
I'm also unsure how to feel about it: on one hand, I'm tempted to just make `from_json` return `None` when the file does not exist; OTOH, it would feel very much like swallowing errors, and if I see a `try_from_json` I would likely need to open its definition (or documentation) to know what it does.

Reply:
that's fair. let's just leave this part as-is for now. :slightly_smiling_face: